pkg/module/build.go

```diff
@@ -90,7 +90,15 @@ func CreateArchive(fs vfs.FileSystem, path, gitRemote string, def *Definition, i
 func addSources(ctx cpi.Context, cd *ocm.ComponentDescriptor, def *Definition, gitRemote string) error {
 	if strings.HasSuffix(def.Source, ".git") {
 		gitSource := gitsource.NewGitSource()
-		src, err := gitSource.FetchSource(ctx, def.Source, def.Repo, def.Version, gitRemote)
+		if def.Repo == "" {
+			var repo string
+			var err error
+			if repo, err = gitSource.DetermineRepositoryURL(gitRemote, def.Source); err != nil {
```

Comment:
I like you intended to extract this `DetermineRepositoryURL` out of `FetchSource`, but please consider rewrite this condition and value assign a bit shorter.

for example, you can try to refactor this `DetermineRepositoryURL` to accept `def.Repo` as parameter, directly return and assign to def.Repo if it's not empty.
```
var err error
if def.Repo, err = gitSource.DetermineRepositoryURL(gitRemote, def.Repo, def.Source); err != nil {
  return err
}
```

Reply:
can you verify if htis def.Repo have value? by looking at the [place](https://github.com/kyma-project/cli/blob/dfbb1aa16491bc5b8e9f5a722c37f3ddd6e450bd/pkg/module/resources.go#L252) it get assigned, seems that `p.Repo` also an empty one.

If it's true, then we can even remove this `def.Repo`

Reply:
after checking this PROJECT file, I think we don't need this `def.Repo`, basically
1. The `Repo` is the repository this module repository, the user run against kubebuilder will anyway run CLI inside module repo, it makes no sense to read it from `PROJECT` file.
2. After remove this `def.Repo`, we are make sure the `DetermineRepositoryURL` use repoPath as the single source of truth.
3. This kubebuilder will deprecated soon, this `def.Repo` will removed anyway.